app/api/v2/identity/sessions.rb

```diff
@@ -87,6 +93,41 @@ class Sessions < Grape::API
           session.destroy
           status(200)
         end
+
+        namespace :auth0 do
+          desc 'Auth0 authentication by id_token',
+               success: { code: 200, message: 'User authenticated' },
+               failure: [
+                 { code: 400, message: 'Required params are empty' },
+                 { code: 404, message: 'Record is not found' }
+               ]
+          params do
+            requires :id_token,
+                     type: String,
+                     desc: 'id_token from Auth0'
+          end
+          post '/auth' do
+            claims = Barong::Auth0.new.decode_and_verify(params[:id_token])
+            status(404) unless claims.key?('email')
+
+            user = get_user(claims['email'])
+            csrf_token = open_session(user)
+            publish_session_create(user)
+
+            present user, with: API::V2::Entities::UserWithFullInfo, csrf_token: csrf_token
+            status(200)
+          end
+
+          desc 'Logout from Auth0',
+               success: { code: 302, message: 'User logged out' },
+               failure: [
+                 { code: 404, message: 'Record is not found' }
+               ]
+          get '/logout' do
+            uri = Barong::Auth0.new.logout_uri(Barong::App.config.auth0_client_id, Barong::App.config.auth0_logout_uri)
+            redirect uri
```

Comment:
we should destroy session in barong too

and don't need redirect

Reply:
I think we don't need the logout form auth0 in Barong. The logout should belong to FE. Just session destroy is enough.